packages/admin/admin/src/mui/menu/CollapsibleItem.styles.ts

```diff
@@ -0,0 +1,68 @@
+import { Theme } from "@mui/material";
+import { createStyles } from "@mui/styles";
+
+import { MenuCollapsibleItemProps } from "./CollapsibleItem";
+
+export type MenuCollapsibleItemClassKey =
+    | "root"
+    | "childSelected"
+    | "listItem"
+    | "open"
+    | "collapsedMenuParentTitle"
+    | "itemTitle"
+    | "collapsibleIcon"
+    | "colorGrey"
+    | "colorWhite"
+    | "collapsibleOpen";
+
+export const styles = (theme: Theme) =>
+    createStyles<MenuCollapsibleItemClassKey, MenuCollapsibleItemProps>({
+        root: {},
+        childSelected: {
+            color: theme.palette.primary.main,
+            "& $listItem": {
+                "& [class*='MuiListItemText-root']": {
+                    color: theme.palette.primary.main,
+                    "& [class*='MuiListItemText-primary']": {
+                        fontWeight: ({ level }) => level === 2 || (level === 3 && theme.typography.fontWeightBold),
+                    },
+                },
+                "& [class*='MuiListItemIcon-root']": {
+                    color: theme.palette.primary.main,
+                },
+            },
+            "& [class*='MuiListItemIcon-root']": {
+                color: theme.palette.primary.main,
+            },
+        },
+        collapsedMenuParentTitle: {
+            backgroundColor: ({ level, isMenuOpen, isCollapsibleOpen }) =>
+                level === 1 && !isMenuOpen && isCollapsibleOpen ? theme.palette.primary.main : theme.palette.grey[50],
+        },
+        itemTitle: {
+            fontWeight: theme.typography.fontWeightBold,
+            fontSize: 12,
+            padding: "15px 15px 20px 15px",
+            lineHeight: "16px",
+            color: theme.palette.grey[500],
+        },
+        collapsibleOpen: {
+            backgroundColor: ({ isMenuOpen }) => (!isMenuOpen ? theme.palette.primary.main : undefined),
+            color: ({ isMenuOpen }) => (!isMenuOpen ? `${theme.palette.common.white} !important` : undefined),
+            "& [class*='MuiListItemIcon-root']": {
+                color: ({ isMenuOpen }) => (!isMenuOpen ? `${theme.palette.common.white} !important` : undefined),
+            },
+        },
+        collapsibleIcon: {
+            fontSize: 12,
+            color: theme.palette.grey[200],
+        },
+        colorGrey: {
+            color: theme.palette.grey[200],
+        },
+        colorWhite: {
+            color: theme.palette.common.white,
+        },
```

Comment:
The class-key should make it understandable, which element you are styling and in which state. 
Maybe they should be named something like `collapsibleIconLevelOne` and `collapsibleIconLevelTwo` rather than naming them after the styling they contain.

Reply:
✅